pkg/kubelet/pleg/generic.go

```diff
@@ -488,85 +476,30 @@ func (g *GenericPLEG) updateCache(ctx context.Context, pod *kubecontainer.Pod, p
 	return status, g.cache.Set(pod.ID, status, err, timestamp), err
 }
 
-// SetPodWatchCondition flags the pod for reinspection on every Relist iteration until the watch
-// condition is met. The condition is keyed so it can be updated before the condition
-// is met.
-func (g *GenericPLEG) SetPodWatchCondition(podUID types.UID, conditionKey string, condition WatchCondition) {
-	g.watchConditionsLock.Lock()
-	defer g.watchConditionsLock.Unlock()
-
-	conditions, ok := g.watchConditions[podUID]
-	if !ok {
-		conditions = make(map[string]versionedWatchCondition)
-	}
+// RequestPodReSync triggers a syncPod event for the pod by generating a pod event in the PLEG.
+func (g *GenericPLEG) RequestPodReSync(podUID types.UID) {
+	g.podsToTriggerSyncLock.Lock()
+	defer g.podsToTriggerSyncLock.Unlock()
 
-	versioned, found := conditions[conditionKey]
-	if found {
-		// Watch condition was already set. Increment its version & update the condition function.
-		versioned.version++
-		versioned.condition = condition
-		conditions[conditionKey] = versioned
-	} else {
-		conditions[conditionKey] = versionedWatchCondition{
-			key:       conditionKey,
-			condition: condition,
-		}
-	}
-
-	g.watchConditions[podUID] = conditions
+	g.podsToTriggerSync[podUID] = true
 }
 
-// getPodWatchConditions returns a list of the active watch conditions for the pod.
-func (g *GenericPLEG) getPodWatchConditions(podUID types.UID) []versionedWatchCondition {
-	g.watchConditionsLock.Lock()
-	defer g.watchConditionsLock.Unlock()
+func (g *GenericPLEG) cancelPodReSync(podUID types.UID) {
```

Comment:
nit: maybe call this `completePodResync`? We're not really _canceling_ it, we're just marking it as done